tests/hardware_tests/sensors/Ultrasonic_test.cpp

```diff
@@ -46,44 +47,57 @@ BOOST_AUTO_TEST_SUITE(UltrasonicTest)
         const int echoPin = 24;
         UltrasonicSensor sensor(triggerPin, echoPin);
 
-        
+
         // First, simulate an initial distance measurement to set lastDistance
         // This simulates the sensor being at a stable position before the test begins
         const uint32_t initStartTick = 0;
         const uint32_t initEndTick = 500;            // Arbitrary initial distance
         simulateCallback(echoPin, 1, initStartTick); // Rising edge for initial distance
         simulateCallback(echoPin, 0, initEndTick);   // Falling edge for initial distance
-        
+
         // Calculate the initial distance to set lastDistance inside the sensor
         sensor.calculateDistance();
         sensor.isMotionDetected(5.0f);
-    //    printf("Distance: %f\n", sensor.calculateDistance());
+        //    printf("Distance: %f\n", sensor.calculateDistance());
+
 
-        
         // Simulate no motion triggers warning system
         const uint32_t startTick1 = 0;
         const uint32_t endTick1 = 583; // Time for 10 centimeters in microseconds (582.75us)
 
         simulateCallback(echoPin, 1, startTick1);
         simulateCallback(echoPin, 0, endTick1);
 
-        // Check if motion was detected
+        // Check if motion was not detected
         bool motionDetected = sensor.isMotionDetected(5.0f);
-    //    printf("Motion detected: %s\n", motionDetected ? "TRUE" : "FALSE");
+        //    printf("Motion detected: %s\n", motionDetected ? "TRUE" : "FALSE");
         BOOST_TEST(!motionDetected, "Motion should not be detected");
 
-        
+
         // Simulate motion under the threshold set to trigger warning system
         const uint32_t startTick2 = 0;
         const uint32_t endTick2 = 291; // Time for 5 centimeters in microseconds (291.375us)
 
         simulateCallback(echoPin, 1, startTick2);
         simulateCallback(echoPin, 0, endTick2);
 
-        // Check if motion was detected
-        motionDetected = sensor.isMotionDetected(5.0f);
-    //    printf("Motion detected: %s\n", motionDetected ? "TRUE" : "FALSE");
-        BOOST_TEST(motionDetected, "Motion should be detected");
+////         Check if motion was detected
```

Comment:
@AlexandruRO45 I need your help with this

Reply:
Solved by modifying isMotionDetected() function:

- check if the absolute difference between the current distance and the last distance is greater than or equal to the distanceThreshold, and the current distance is less than 25.0f.
- if the distance change exceeds the threshold, we check if the time since the last motion detection event is greater than or equal to the debounceMs value (500 milliseconds by default).
- if the time since the last motion detection event exceeds the debounceMs threshold, we set motionDetected to true and update the lastMotionDetectedAt variable with the current time.

This way, the sensor will only trigger a motion detection event if the distance change exceeds the threshold and a certain amount of time (the debounce time) has elapsed since the last motion detection event. This helps to filter out temporary, sporadic distance changes and improve the reliability of the motion detection.